src/seedu/addressbook/commands/Command.java

```diff
@@ -63,6 +63,14 @@ protected ReadOnlyPerson getTargetPerson() throws IndexOutOfBoundsException {
         return relevantPersons.get(getTargetIndex() - DISPLAYED_INDEX_OFFSET);
     }
 
+    /**
+     * @return true for command types that mutate the data. Otherwise, return false.
+     * Returns false by default.
+     */
+    public boolean isMutating() {
+        return false;
```

Comment:
You can remove `@return`, instead write it like this `Returns false if …`

Having a default implementation for the `isMutating` method helps to reduce the amount of code. However, some may argue that choosing `false` (or `true`) as the default is arbitrary and hurts the intuitiveness of the code. Making the method `abstract` forces all child classes to declare whether it is a mutating command or not, rather than depend on an arbitrary decision made in the parent class.